biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -357,6 +360,47 @@ public void analyze() throws Exception {
 						uses.transpose().get(Descriptors.DEFAULT_PACKAGE));
 			}
 
+			// Check for use of the deprecated bnd @Export annotation
+
+			TypeRef bndAnnotation = descriptors.getTypeRefFromFQN(aQute.bnd.annotation.Export.class.getName());
+			contained.keySet()
+					.stream()
+					.map(this::getPackageInfoClazz)
+					.filter(clz -> clz != null)
+					.filter(clz -> clz.annotations != null)
+					.filter(clz -> clz.annotations.contains(bndAnnotation))
+					.map(Clazz::getClassName)
+					.map(TypeRef::getPackageRef)
+					.map(PackageRef::getFQN)
+					.forEach(fqn -> warning(
+							"The annotation aQute.bnd.annotation.Export applied to package %s is deprecated and will be removed in a future release. The org.osgi.annotation.bundle.Export should be used instead",
+							fqn));
+		}
+	}
+
+	private Parameters getExportedByAnnotation() {
+		TypeRef exportAnnotation = descriptors.getTypeRefFromFQN("org.osgi.annotation.bundle.Export");
+		return contained.keySet()
+				.stream()
+				.map(this::getPackageInfoClazz)
+				.filter(clz -> clz != null)
+				.filter(clz -> clz.annotations != null)
+				.filter(clz -> clz.annotations.contains(exportAnnotation))
+				.map(Clazz::getClassName)
+				.map(TypeRef::getPackageRef)
+				.map(PackageRef::getFQN)
+				.reduce((a, b) -> a + "," + b)
```

Comment:
`.collect(Collectors.joining(","))`

Or better yet, modify Parameters to add a method which returns a Collector you can use here. See <https://github.com/bndtools/bnd/blob/5eee8ed041f0eb9e089590c1d257d43c97a7317b/biz.aQute.bndlib/src/aQute/bnd/osgi/repository/ResourcesRepository.java#L66-L68> for example.